cmd/demo/demo.go

```diff
@@ -50,13 +56,42 @@ func runDemo(c *cobra.Command, args []string) {
 	if flags.testAPIEnabled {
 		StartTestAPI()
 	}
-	p := prompt.New(
-		executor,
-		completer,
-		prompt.OptionPrefix("> "),
-		prompt.OptionTitle("perun"),
-	)
-	p.Run()
+	if flags.useStdIO {
+		runWithStdIO(executor)
+	} else {
+		p := prompt.New(
+			executor,
+			completer,
+			prompt.OptionPrefix("> "),
+			prompt.OptionTitle("perun"),
+		)
+		p.Run()
+	}
+}
+
+func runWithStdIO(executor func(string)) {
+	stdinChannel := newStdinChannel()
+	for {
```

Comment:
If you are ok with omitting the `printf` you could do:  
```go
func runWithStdIO(executor func(string)) {
	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		executor(scanner.Text())
	}
	os.Exit(1)
}
```
the execution is synchronous right now anyway, since the input channel is unbuffered.  
Then you could remove `newStdinChannel`.

Reply:
ok, done